frontend/src/lib/components/LemonCalendar/LemonCalendarRange.tsx

```diff
@@ -0,0 +1,146 @@
+import { formatDateRange } from 'lib/utils'
+
+export interface LemonCalendarRangeProps {
+    value?: (string | null)[]
+    onChange: (date: string[]) => void
+    months?: number
+    getLemonButtonProps?: LemonCalendarProps['getLemonButtonProps']
+    onClose?: () => void
+}
+
+export function LemonCalendarRange({ value, onChange, onClose, months }: LemonCalendarRangeProps): JSX.Element {
+    // Keep a sanitised and cached copy of the selected range
+    const [valueStart, valueEnd] = [
+        value?.[0] ? dayjs(value[0]).format('YYYY-MM-DD') : null,
+        value?.[1] ? dayjs(value[1]).format('YYYY-MM-DD') : null,
+    ]
+    const [[rangeStart, rangeEnd], setRange] = useState([valueStart, valueEnd])
+
+    // Track if the last change was on the range's start or end
+    const [lastChanged, setLastChanged] = useState('end' as 'start' | 'end')
+    useEffect(() => {
+        console.log({ valueStart, rangeStart, valueEnd, rangeEnd })
+        if (valueStart !== rangeStart) {
+            setLastChanged('start')
+        }
+        if (valueEnd !== rangeEnd) {
+            setLastChanged('end')
+        }
+    }, [valueStart, rangeStart, valueEnd, rangeEnd])
+
+    // How many months fit on the screen, capped between 1..2
+    function getMaxMonthCount(): number {
+        const width = typeof window === undefined ? 1024 : window.innerWidth
+        return Math.min(Math.max(1, Math.floor(width / 300)), 2)
+    }
+    const [autoMonthCount, setAutoMonthCount] = useState(getMaxMonthCount())
+    useEffect(() => {
+        const listener = (): void => {
+            if (autoMonthCount !== getMaxMonthCount()) {
+                setAutoMonthCount(getMaxMonthCount())
+            }
+        }
+        window.addEventListener('resize', listener)
+        return () => window.removeEventListener('resize', listener)
+    }, [autoMonthCount])
+
+    // What months exactly are shown on the calendar
+    const shownMonths = months ?? autoMonthCount
+    const rangeMonthDiff =
+        rangeStart && rangeEnd ? dayjs(rangeEnd).startOf('month').diff(dayjs(rangeStart).startOf('month'), 'month') : 0
+    const firstMonthForRange = dayjs(rangeStart ?? rangeEnd ?? undefined)
+        .subtract(Math.max(0, shownMonths - 1 - rangeMonthDiff), 'month')
+        .startOf('month')
+        .format('YYYY-MM-DD')
+    const [firstMonth, setFirstMonth] = useState(firstMonthForRange)
+
+    // If the range changes via props and is not in view, update the first month
+    useEffect(() => {
+        const lastMonthForRange = dayjs(firstMonthForRange)
+            .add(shownMonths - 1, 'month')
+            .endOf('month')
+        if (
+            rangeStart &&
+            rangeEnd &&
+            (dayjs(rangeStart).isAfter(lastMonthForRange) || dayjs(rangeEnd).isBefore(dayjs(firstMonthForRange)))
+        ) {
+            setFirstMonth(firstMonthForRange)
+        }
+    }, [rangeStart, rangeEnd])
+
+    return (
+        <div className="LemonCalendarRange">
```

Comment:
Weirdly the sizing of buttons changes in various ways when I make a selection:

![2022-08-31 13 09 20](https://user-images.githubusercontent.com/4550621/187671090-6a6e9400-e6ba-4ba6-9ad6-4d4652146639.gif)

As a side note, it'd look amazing if a selection looked contiguous instead of the buttons being highlighted individually… though this might be hard to pull off with an HTML table

Reply:
I got rid of it when I added a 40px width to each cell. It looks nicer with square buttons as well

<img width="705" alt="image" src="https://user-images.githubusercontent.com/53387/187696959-034635a6-90f9-48c0-8a51-f54e743f1853.png">